privx_api/base.py

```diff
@@ -12,7 +12,10 @@
 
 
 def format_path_components(format_str: str, **kw) -> str:
-    components = {k: urllib.parse.quote(v, safe="") for k, v in kw.items()}
+    try:
+        components = {k: urllib.parse.quote(v, safe="") for k, v in kw.items()}
+    except TypeError:
+        raise InternalAPIException("Incorrect type of the passed arguments")
```

Comment:
Could be more explicit about what params had violated the constraint. A quick example to consider:
```
incorrect_params = [k: f"{type(v)}" for k, v in kw.items() if isintance(v, str) is False]
error_message = f"Incorrect ...:\n{json.dumps(incorrect_params, indent=4)}"
raise InternalAPIException(error_message)
```